core/templates/dev/head/pages/learner_dashboard/LearnerDashboard.js

```diff
@@ -191,6 +227,100 @@ oppia.controller('LearnerDashboard', [
       return value;
     };
 
+    $scope.sortFeedbackFunction = function(entity) {
+      return entity[$scope.currentFeedbackThreadsSortType];
+    }
+
+    $scope.onClickThread = function(
+      explorationId, threadId, explorationTitle) {
+      var threadDataUrl = (
+        '/learnerdashboardthreadhandler/' + explorationId + '/' + threadId);
+      $scope.threadTitle = explorationTitle;
+      $scope.feedbackThreadActive = true;
+      $scope.explorationId = explorationId;
+      $scope.threadId = threadId;
+
+      for (var index = 0; index < $scope.threadSummaries.length; index++) {
+        if ($scope.threadSummaries[index].exploration_id === explorationId &&
+            $scope.threadSummaries[index].thread_id === threadId) {
+          $scope.threadIndex = index;
+          var threadSummary = $scope.threadSummaries[index]
+          if (typeof threadSummary.second_last_message_read !== 'undefined') {
+            $scope.threadSummaries[index].second_last_message_read = true;
+          }
+          if (threadSummary.last_message_read === false) {
+            $scope.numberOfUnreadThreads -= 1;
+          }
+          threadSummary.last_message_read = true;
+        }
+      }
+
+      $http.get(threadDataUrl).then(function(response) {
+        var data = response.data;
+        $scope.messageSummaries = data.message_summary_list;
+      });
+    };
+
+    $scope.backFromThread = function() {
+      $scope.feedbackThreadActive = false;
+    };
+
+    $scope.addNewMessage = function(explorationId, threadId, newMessage) {
+      var url = '/threadhandler/' + explorationId + '/' + threadId;
+      var payload = {
+        updated_status: null,
+        updated_subject: null,
+        text: newMessage
+      };
+      $scope.messageSendingInProgress = true;
+      $http.post(url, payload).success(function() {
+        var messageSummary = {};
```

Comment:
Could you please explain this a bit further? Here I've just temporarily updated the summaries so that if the user clicks return to feedback threads he/she is able to see updated fields. If the user reloads the learner dashboard the summaries come naturally as we have sent a post request above having a payload that only contains the new message.